### PR TITLE
Nav 38 data loader

### PR DESCRIPTION
**Pull request by jsoppela**
Tools to initialise app database with graphs defined in spreadsheets

### CODE REVIEW

File: navigator_engine/initial_data_loader.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    model.db.session.add(graph)
+    model.db.session.commit()
+
+    # Reload the graph fromt he db and check that the title of the root node is updated
+    graph_reloaded = model.Graph.query.filter_by(id=1).first()
+    graph_reloaded.to_networkx()
+    root = [n for n, d in graph_reloaded.network.in_degree() if d == 0]
+    root = root[0]
+    current_app.logger.info(f"Reloaded root node {root} with title {root.conditional.title}")
+    assert new_title == root.conditional.title
```
Comment (jonathansberry) [inline navigator_engine/initial_data_loader.py:153]: This function shouldn't be part of the data loader.  It is already included in the testing code. I think you can just delete it.

File: navigator_engine/initial_data_loader.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                    data_dict[row[0]][data_headers[i]] = row[i]
+            row_no = row_no + 1
+
+    # TODO: validate the imported graph CSV
+
+    # Load a simple BDG
+    graph = model.Graph(
+        title=milestone_dict['Milestone Title'],
+        version=milestone_dict['Version'],
+        description=milestone_dict['Description'])
```
[the remark this answers is not in the document]
Comment (jonathansberry) [reply navigator_engine/initial_data_loader.py:56]: Can you do the same for lines 81, 106, 115, 121

**Review by jonathansberry — COMMENTED** [review]
Thanks for the great start on this work. Aware you still have further work planned for this.  Here are some comments to help you along...

**Comment by jonathansberry** [comment]
Actually can we call the module filename "graph_loader.py"
(a) I don't want to confuse graphs with the "data" that the graphs refer to to make decisions.
(b) The logic is not just for "initial" loading but may be used for loading later as well.

File: navigator_engine/initial_data_loader.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+
+def create_demo_data():
+    # Clear and reset the db
+    model.db.drop_all()
+    model.db.create_all()
+
+    # Read graph data into memory
+    with open(current_app.config.get('GRAPH_CSV'), newline='') as csvfile:
+        data_file = csv.reader(csvfile, delimiter=';')
```
[the remark this answers is not in the document]
Comment (jsoppela) [reply navigator_engine/initial_data_loader.py]: Excel parsing has now been implemented in commit https://github.com/fjelltopp/navigator_engine/pull/4/commits/2d9e37d71c51e589ac53fce429b69fc79a890f7d

**Pull request merged**